fabtests/multinode/src/harness.c

```diff
@@ -57,7 +57,23 @@ static int parse_caps(char* caps)
 	}
 }
 
-static inline ssize_t socket_send(int sock, void *buf, size_t len, int flags)
+static int parse_pattern(char *pattern)
+{
+	if (strcmp(pattern, "full_mesh") == 0) {
+		return 0;
+	} else if (strcmp(pattern, "ring") == 0) {
+		return 1;
+	} else if (strcmp(pattern, "gather") == 0) {
+		return 2;
+	} else if (strcmp(pattern, "broadcast") == 0) {
+		return 3;
+	} else {
+		printf("Warn: Invalid pattern, defaulting to full_mesh\n");
+		return 0;
+	} 
```

Comment:
As a follow up change, please convert the pattern to an enum, so we don't need a magic decoder ring to determine that '2' is gather.  E.g.
```
enum mnode_pattern {
    PATTERN_MESH,
    PATTERN_RING,
    ...
};

enum mnode_patter pattern;
```